flytekit/image_spec/image_spec.py

```diff
@@ -228,15 +230,16 @@ def build(cls, image_spec: ImageSpec) -> str:
             click.secho(f"Image {img_name} not found. Building...", fg="blue")
             if builder not in cls._REGISTRY:
                 raise Exception(f"Builder {builder} is not registered.")
-            if builder == "envd":
-                envd_version = metadata.version("envd")
-                # flytekit v1.10.2+ copies the workflow code to the WorkDir specified in the Dockerfile. However, envd<0.3.39
-                # overwrites the WorkDir when building the image, resulting in a permission issue when flytekit downloads the file.
-                if Version(envd_version) < Version("0.3.39"):
-                    raise FlyteAssertion(
-                        f"envd version {envd_version} is not compatible with flytekit>v1.10.2."
-                        f" Please upgrade envd to v0.3.39+."
-                    )
+            with contextlib.suppress(PackageNotFoundError):
+                if builder == "envd":
+                    envd_version = metadata.version("envd")
+                    # flytekit v1.10.2+ copies the workflow code to the WorkDir specified in the Dockerfile. However, envd<0.3.39
+                    # overwrites the WorkDir when building the image, resulting in a permission issue when flytekit downloads the file.
+                    if Version(envd_version) < Version("0.3.39"):
```

Comment:
not sure i follow... if the exception is suppressed, why have it at all?

Reply:
is this to handle the case someone uses `Imagespec` but don't have envd installed? What happens in that case? What should happen?

Reply:
Looking this over, I think this PR is not required anymore.

When `builder=="envd"`, `envd` should be installed at that point.